### PR TITLE
Properly crop images in avatars API

### PR DESCRIPTION
**Pull request by Meldiron**
## What does this PR do?

Solves issue when resizing was not done properly if only one parameter was provided.

- [x] Fix Initials
- [x] Fix Flags
- [x] Fix Credit Cards
- [x] Fix Browsers
- [x] Fix URL Image

Not only we support proper resizing now, we also return full resolution unless asked to downscale.

## Test Plan

- [x] Manual QA

CleanShot 2022-04-12 at 13 36 30
CleanShot 2022-04-12 at 13 37 48
CleanShot 2022-04-12 at 13 37 31
CleanShot 2022-04-12 at 13 37 08

(Yes, I tested all affected endpoints)

## Related PRs and Issues

https://github.com/appwrite/appwrite/issues/3082

### Have you read the [Contributing Guidelines on issues](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md)?

Yes

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1202115796713529

### CODE REVIEW

**Comment by eldadfux** [comment]
Not sure about this.

1. this is a breaking change
2. default image will have 0x0 size
3. If this is the case, should we make both params required?

Could this be solved by just mentioning that providing the value 0 will keep the original width or height?

**Comment by Meldiron** [comment]
@eldadfux 

1. Kind-of. Same code will still work, but instead of getting default size of 100x100, you will get source image at maximum resolution
2. Correct and that's fine, that means original resolutions will be provided

Documenting endpoints better would also be a valid solution, yes.

**Comment by eldadfux** [comment]
I think I would prefer to force the 100x100 resolution for preview images, to make devs aware of this instead of loading images in full resolution by default where most of the time it's far from optimized.

**Comment by Meldiron** [comment]
@eldadfux PR updated to only address documentation changes. It also includes fox when QR code size 0x0 was allowed, resulting in a server error.

File: docs/references/avatars/get-browser.md
Comment (eldadfux) [inline docs/references/avatars/get-browser.md:1]:
```suggestion
You can use this endpoint to show different browser icons to your users. The code argument receives the browser code as it appears in your user /account/sessions endpoint. Use width, height and quality arguments to change the output settings.
```
We should / could use a link for `/account/sessions`?

**Review by eldadfux — COMMENTED** [review]
One minor comment.

**Review by eldadfux — APPROVED** [review]

**Comment by eldadfux** [comment]
Approved, once conflict is resolved we can merge.

**Pull request merged**
